src/conversions.rkt

```diff
@@ -33,29 +33,31 @@
 
   (unless (hash-has-key? parametric-operators conv1)
     (define impl (compose (representation-bf->repr repr2) (representation-repr->bf repr1)))
-    (register-operator! conv1 conv1 (list prec1) prec2  ; fallback implementation
-      (list (cons 'fl impl) (cons 'bf identity)
-            (cons 'ival identity) (cons 'nonffi impl))))
+    (register-operator! conv1 (list (cons 'bf identity) (cons 'ival identity) (cons 'nonffi impl)))
+    (register-operator-impl! conv1 conv1 (list prec1) prec2  ; fallback implementation
+      (list (cons 'fl impl))))
   
   (unless (hash-has-key? parametric-operators conv2)
     (define impl (compose (representation-bf->repr repr1) (representation-repr->bf repr2)))
-    (register-operator! conv2 conv2 (list prec2) prec1  ; fallback implementation
-      (list (cons 'fl impl) (cons 'bf identity)
-            (cons 'ival identity) (cons 'nonffi impl))))
+    (register-operator! conv2 (list (cons 'bf identity) (cons 'ival identity) (cons 'nonffi impl)))
+    (register-operator-impl! conv2 conv2 (list prec2) prec1  ; fallback implementation
+      (list (cons 'fl impl))))
```

Comment:
Perhaps what we should have is a general-purpose operator called `cast` which then has an implementation for each conversion?

Reply:
That would be nice. The problem is the abstract name is also the resugared name, so we'd lose type info when resugaring. Are we going to need a 3rd field? Something like  `register-operator-impl! resugared-name  abstract-name repr-name ...`

Reply:
Ah, ok, then let's not bother for now.

Reply:
Alternatively, we can resugar into proper FPCore precision annotations.